packages/ibm-products/src/components/TagSet/TagSet.tsx

```diff
@@ -128,9 +121,9 @@ export interface TagSetProps extends PropsWithChildren {
   /**
    * label for the overflow show all tags link.
    *
-   * **Note:** Required if more than 10 tags
+   *  **Note: Required if more than 10 tags**
    */
-  showAllTagsLabel: string;
+  showAllTagsLabel?: string;
```

Comment:
Hey @AlexanderMelox ,
Approach looks good. 
But, where are we specifying the default label for `showAllTagsLabel`?

Reply:
@amal-k-joy so this was a good discussion because in the proptypes it is required if there are more than 10 tags present. There is currently no way to represent this logic with typescript types. Initially In this PR I tried to add default string parameters but it was making test run for ever and timeout, you can see it in [this commit ](https://github.com/carbon-design-system/ibm-products/pull/5962/commits/10a4663767afbf392d2ba6b73b9b172fedbc80b2)

```
 /**
   * label for the overflow show all tags link.
   *
   * **Note:** Required if more than 10 tags
   */
  showAllTagsLabel: string_required_if_more_than_10_tags,
```

Reply:
@davidmenendez later found out that the default string params were okay but the `tags = []` was causing the issue. So I will be adding it back. Thanks @amal-k-joy

Reply:
I removed requirements to add props after 10 tags or more and just added optional props and default strings. I think this is a better approach and we do this in other components anyways.

Reply:
Hey @AlexanderMelox , 
This feels like a better approach since it make more easier for adopters. But can we update this prop description as well  ? it still says the props as required.
![image](https://github.com/user-attachments/assets/7f3a47b1-63e0-47b7-b395-ad06c58e1a97)

Reply:
Hi @AlexanderMelox ,
This feels like better approach since this make it more easier for users. 
One minor thing. Storybook props description need to be updated accordingly.

![image](https://github.com/user-attachments/assets/ddf1bf1d-aeef-423d-9d9d-537fc1914aaa)

Reply:
Done ✅  thank you for the feedback